src/components/CardEditor/CardEditForm/CardEditForm.jsx

```diff
@@ -27,67 +56,517 @@ const propTypes = {
   }),
 };
 
-const CardEditForm = ({ value, /* errors, */ onChange, i18n }) => {
+const CardEditForm = ({ value, /* errors, */ onChange, /* onAddImage, */ i18n }) => {
   const mergedI18N = { ...defaultProps.i18n, ...i18n };
   const [showEditor, setShowEditor] = useState(false);
   const [modalData, setModalData] = useState();
 
-  // temporary JSON view
-  const tempJSONEditor = (
-    <Modal
-      className="temp-json-editor"
-      modalHeading="Edit JSON"
-      open={showEditor}
-      onClose={() => setShowEditor(false)}
-      onRequestClose={() => setShowEditor(false)}
-      onRequestSubmit={() => {
-        setShowEditor(false);
-        onChange(JSON.parse(modalData));
-      }}
-      primaryButtonText="Save"
-      secondaryButtonText="Cancel"
-    >
-      <TextArea
-        labelText="Card data"
-        rows={8}
-        onChange={evt => setModalData(evt.target.value)}
-        value={modalData}
-      />
-    </Modal>
-  );
-
   const baseClassName = `${iotPrefix}--card-edit-form`;
 
+  const allowedSizesForType = {
```

Comment:
It would be better if we could pull these from the components themselves in case they change in the future

Reply:
Yep-- we'd need some guidance from design as to what sizes we permit.  I just inferred this list from looking at what the cards "gracefully" handle.  I'll revisit this in the next PR.

Reply:
Added these as a constant to LayoutConstants